nixos/modules/services/web-apps/alist.nix

```diff
@@ -0,0 +1,295 @@
+                default = "";
+                description = ''
+                  To control the encryption options during the SSL handshake, the parameters can be searched by themselves, or check the answer from ChatGPT on https://alist.nn.ci/config/configuration.html#database
+                '';
+              };
+              dsn = lib.mkOption {
+                type = lib.types.str;
+                default = "";
+                description = ''
+                  see https://github.com/alist-org/alist/pull/6031
+                '';
+              };
+            };
+            scheme = {
+              address = lib.mkOption {
+                type = lib.types.str;
+                default = "0.0.0.0";
+                description = ''
+                  The http/https address to listen on, default `0.0.0.0`
+                '';
+              };
+              http_port = lib.mkOption {
+                type = lib.types.oneOf [
+                  (lib.types.enum [ (-1) ])
+                  lib.types.port
+                ];
+                default = 5244;
+                description = ''
+                  The http port to listen on, default `5244`, if you want to disable http, set it to `-1`
+                '';
+              };
+              https_port = lib.mkOption {
+                type = lib.types.oneOf [
+                  (lib.types.enum [ (-1) ])
+                  lib.types.port
+                ];
+                default = (-1);
+                description = ''
+                  The https port to listen on, default `-1`, if you want to enable https, set it to non `-1`
+                '';
+              };
+              force_https = lib.mkOption {
+                type = lib.types.bool;
+                default = false;
+                description = ''
+                  Whether HTTPS protocol is forced, if true the user can only access the website through HTTPS.
+                '';
+              };
+              cert_file = lib.mkOption {
+                type = lib.types.str;
+                default = "";
+                description = "Path of cert file";
+              };
+              key_file = lib.mkOption {
+                type = lib.types.str;
+                default = "";
+                description = "Path of key file";
+              };
+              unix_file = lib.mkOption {
+                type = lib.types.str;
+                default = "";
+                description = ''
+                  Unix socket file path to listen on, default empty, if you want to use unix socket, set it to non empty
+                '';
+              };
+              unix_file_perm = lib.mkOption {
+                type = lib.types.str;
+                default = "";
+                description = ''
+                  Unix socket file permission, set to the appropriate permissions
+                '';
+              };
+            };
+            temp_dir = lib.mkOption {
```

Comment:
I think this can be removed and you can use `PrivateTmp = true;`

Reply:
You mean hardcode the temp path to `/tmp`?

Reply:
I would prefer to not specify that and let systemd provide you the path (`%T`), see: https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html

Reply:
We should specific this part because `services.alist.settings` is actually used to generate config file.

Reply:
Mmh OK.